src/structures/Invite.js

```diff
@@ -20,14 +20,13 @@ class Invite extends Base {
 
   _patch(data) {
     const InviteGuild = require('./InviteGuild');
-    const Guild = require('./Guild');
     /**
      * The guild the invite is for including welcome screen data if present
      * @type {?(Guild|InviteGuild)}
      */
     this.guild = null;
     if (data.guild) {
-      this.guild = data.guild instanceof Guild ? data.guild : new InviteGuild(this.client, data.guild);
+      this.guild = this.client.guilds.resolve(data.guild) ?? new InviteGuild(this.client, data.guild);
```

Comment:
Since `GuildManager#resolve` doesn't accept an api guild object:
```suggestion
      this.guild = this.client.guilds.resolve(data.guild.id) ?? new InviteGuild(this.client, data.guild);
```

Reply:
It does and I tested it 🤔

Reply:
I thought I tested this but I was executing the wrong method, I am the entire circus rn

Reply:
`GuildManager#resolve` only checks for specific class instances:
https://github.com/discordjs/discord.js/blob/d70bdf2ec4c8dbf1f840ec9e4cfde206f4d61745/src/managers/GuildManager.js#L107-L118

`DataManager#resolve` then checks for whether the object is an instance of `Guild` or whether it's a string being a key of a cached value:
https://github.com/discordjs/discord.js/blob/d70bdf2ec4c8dbf1f840ec9e4cfde206f4d61745/src/managers/DataManager.js#L39-L43

I don't see how it would have worked in your test, it doesn't work on my end.

Reply:
Yeah sorry for the confusion, my bad